turbopack/crates/turbopack-cli-utils/src/issue.rs

```diff
@@ -170,12 +171,105 @@ pub fn format_issue(
             writeln!(styled_issue, "{path}").unwrap();
         }
     }
+    let traces = &*plain_issue.import_traces;
+    if !traces.is_empty() {
+        /// Returns the layer used by all items in the trace if it is unique.
+        /// Returns None, if there are multiple different layers (or no layers)
+        fn get_layer(items: &[PlainTraceItem]) -> Option<RcStr> {
+            let (unique, value) = items
+                .iter()
+                .map(|i| (true, &i.layer))
+                .reduce(|(unique, prev), (_, next)| {
+                    if unique && prev == next {
+                        (true, prev)
+                    } else {
+                        (false, prev)
+                    }
+                })
+                .expect("traces are never empty");
+            if unique { value.clone() } else { None }
+        }
+        fn format_trace_items(
+            out: &mut String,
```

Comment:
Could this just take a `Write` trait? https://doc.rust-lang.org/std/io/trait.Write.html
Would that be better? Not sure.

Reply:
then i need to add a bunch of `unwrap` calls or `?`s which seems lame...